### PR TITLE
Cs 2162

### PR DESCRIPTION
**Pull request by thdlam**
Scheduled job to automatically update application status when health check has failed or succeeded more than 5 times to either operational or reported issues

### CODE REVIEW

File: apps/status-service/src/app/jobs/checkEndpoint.ts [hunk 1]
```diff
@@ -280,3 +284,58 @@ const updateAppStatus = async (
...
+      const failedForMoreThanFiveMinutes = recentHistory.filter((hist) => !hist.ok && dateDiff >= 5).length > 0;
+      const succeededMoreThanFiveMinutes = recentHistory.filter((hist) => hist.ok && dateDiff >= 5).length > 0;
+      const currentServiceStatus = await serviceStatusRepository.get(app.appKey);
+
+      //Set the status to opperational if it is in report issues status and
+      //the the health check has failed for the last five minutes
+      if (
+        currentServiceStatus &&
+        currentServiceStatus.status !== '' &&
+        currentServiceStatus.status === 'operational' &&
```
Comment (solittlework) [inline apps/status-service/src/app/jobs/checkEndpoint.ts:321]: I think currentServiceStatus?.status === 'operational' will be enough.

File: apps/status-service/src/app/jobs/checkEndpoint.ts [hunk 2]
```diff
@@ -280,3 +284,58 @@ const updateAppStatus = async (
...
+      );
+
+      //get first and last date time to make sure the time is 5 minutes apart.
+      const mostRecentDate = moment(new Date(recentHistory.at(0).timestamp));
+      const oldestDate = moment(new Date(recentHistory.at(-1).timestamp));
+
+      //Need to round date diff as MomentJS can give you an incorrect number because of the milliseconds
+      const dateDiff = Math.round(mostRecentDate.diff(oldestDate, 'minute'));
+
+      const failedForMoreThanFiveMinutes = recentHistory.filter((hist) => !hist.ok && dateDiff >= 5).length > 0;
```
Comment (solittlework) [inline apps/status-service/src/app/jobs/checkEndpoint.ts:312]: Considering this edge case:
5 mins before: ok,
4 min before: ok,
3 min before: failed,
2 min before: failed,
Last min: failed

This has not been specified in the task. We might need to clarify it later.

**Review by solittlework — APPROVED** [review]

File: apps/status-service/src/app/jobs/checkEndpoint.ts [hunk 3]
```diff
@@ -280,3 +284,58 @@ const updateAppStatus = async (
...
+        succeededMoreThanFiveMinutes
+      ) {
+        //update status to operational
+        currentServiceStatus.status = 'operational';
+        serviceStatusRepository.save(currentServiceStatus);
+      }
+    }
+  } catch (err) {
+    logger.error(`checkAndUpdateAutoChangeStatus() has encountered an error: ${err}`);
+  }
```
Comment (rstyan) [inline apps/status-service/src/app/jobs/checkEndpoint.ts:340]: Way to jump right in to our sprint retro suggestions!

File: apps/status-service/src/app/jobs/checkEndpoint.ts [hunk 4]
```diff
@@ -280,3 +284,58 @@ const updateAppStatus = async (
...
+        currentServiceStatus &&
+        currentServiceStatus.status !== '' &&
+        currentServiceStatus.status === 'reported-issues' &&
+        succeededMoreThanFiveMinutes
+      ) {
+        //update status to operational
+        currentServiceStatus.status = 'operational';
+        serviceStatusRepository.save(currentServiceStatus);
+      }
+    }
```
Comment (rstyan) [inline apps/status-service/src/app/jobs/checkEndpoint.ts:337]: Suggestion: refactor for readability.  Something like:
if (currentServiceStatus && currentServiceStatus.status !-- '') {
  checkStatusForApplicationFailure(...)
  checkStatusForApplicationRevival(...)
}

**Review by rstyan — APPROVED** [review]

File: apps/status-service/src/app/jobs/checkEndpoint.ts (hunk 4 above)
Comment (rstyan) [reply apps/status-service/src/app/jobs/checkEndpoint.ts:337]: Err... Pauls suggestion for testing the status is correct.  You only need to check currentServiceStatus?.status.  I can never remember what language I'm in :(

**Pull request merged**